### DISCUSSION TITLE
EMFILE: too many open files

### DISCUSSION THREAD

**Discussion by ArneVanhyfte** [aws/aws-sdk-js#4414] [Q&A]
hello

I am trying to make a project with nuxt 3. I am using aws amplify for the authentication in this project. I make use of the api in a pinia store to have my custom authentication flow. 

Everything works great in local development but whenever I try to build the project for hosting I encounter an  EMFILE: too many open files error. Another person working on this project tried building my branch on his machine as well but the same error. 
The strange thing is that the file it says to open after error is different for both tries.

Mine is: .output\server\node_modules\@aws-sdk\client-textract\node_modules\@aws-sdk\credential-provider-node\node_modules\@aws-sdk\credential-provider-imds\node_modules\ts
lib\package.json'

His is: .output\server\node_modules@aws-sdk\client-textract\node_modules@aws-sdk\credential-provider-node\node_modules@aws-sdk\credential-p
rovider-imds\dist\cjs\fromContainerMetadata.js

Any help is appreciated

**Comment by RanVaknin** [comment]
Hi, 

From the description of your problem it seems like there are some file descriptors that are leaking. Your environment usually has a limit of open file descriptors that can be opened at once and that is probably why you are running into this.

To rule out that this is an SDK can you remove that dependency and test without it?
From googling the issue, lots of Nuxt related issues with the same error came up. I believe this is probably related to Nuxt rather than the SDK.

Additionally if you are using a Unix system you can use the `ulimit` command to change the limits on system resources for the shell, ulimit -n will change the limit of maximum number of open file descriptors. 
For example `ulimit -n 4096` or `ulimit -n 10480`. 

Note that this does not fix the memory leak.
